Updates HeroHeadline

MinamiFunakoshiTR opened the pull request:
### What's in this pull request
- Migrates to Svelte 5
- Cleans up docs   
- Cleans up log in HeroHeadline.svelte 
- Renames `stories` folder to `demo`
- Renames `Hero.svelte` to `HeroHeadline.svelte` for consistency 

#### Changes in `HeroHeadline.svelte`
The major changes in `HeroHeadline.svelte` are the slots-to-snippet migration; while I was at it, I streamlined the use of snippets to cut down on code.

Before, the user had to specify the slot name for `background` vs `inline`. Since the only difference between background vs inline is whether the hero is stacked under the headline, I simplified this by creating a `stacked: true` prop to control whether the hero is background vs inline, and turned the previously named slots into an unnamed snippet handled by `children()`. This makes it so that the user doesn't have to write the `{#snippet ...}` code when inserting a snippet, and also cuts down on the number of props.

Since `HeroHeadline` internally uses  `Byline`, I added props such as `getAuthorPage`, `published`, `updated`, etc. that lets the user customise  the byline.  

#### Documentation changes
The previous version had many examples of inline heroes -- inline video, inline graphic, inline photo, etc. Since we now insert inline hero graphics in the same way as we would background hero graphics, and the only difference is whether `stacked` is `true` or `false`, I streamlined the docs to have just one example of an inline hero, clarifying that you'd add other hero media in the same way.

I generally tried to add more explanations to clarify each example.

### Misc. tweaks in `Byline`
- Makes `updateTime` prop optional and makes downstream changes

### Questions
- I didn't add an ArchieML example here because the docs are pretty long and I'm assuming that most people know how to plug in content from an ArchieML doc into `<HeroHeadline>`. But what do you think? Should we add an ArchieML example as we do in all other components?
- I add the [image block in the unstacked, inline version](https://github.com/reuters-graphics/graphics-components/blob/dcf943a01f3f44fc8e73698fcb589c7166997af8/src/components/HeroHeadline/HeroHeadline.svelte#L176-L190) in the same way we add images to the background hero version. I wonder if we should use `FeaturePhoto` for the inline hero instead, since there's no reason why this inline hero needs to be a backgroud-image. It is cleaner code-wise, though, to keep the logic consistent. What do you think? Are you fine with the inline hero image using `background-image`, or do you want it rendered with `FeaturePhoto`?
- Since I renamed `Hero` to `HeroHeadline`, do I have to change the [export in index.js](https://github.com/reuters-graphics/graphics-components/blob/mf-hero/src/index.js#L23)? When I tried pushing this change, git didn't recognise it -- it said nothing to commit, working tree cleaan.

hobbes7878 reviewed (changes requested):
Few things.

1. The SCSS examples in the docs need to be set using `:global()` operators or in `global.scss`, otherwise Svelte will scrub those as unused styles because they don't correspond to an HTML element class within the same context as the component.

2. The docs say to set `stacked` to `true` for inline placement. Think you meant `false`. So `stacked` is by default `true` which overlays the headline on top of the hero media.

3. Let's tag Prasanta to also review the code changes to this component. Would like him to confirm we haven't lost any functionality.

MinamiFunakoshiTR commented:
> 1. The SCSS examples in the docs need to be set using `:global()` operators or in `global.scss`, otherwise Svelte will scrub those as unused styles because they don't correspond to an HTML element class within the same context as the component.

Moved styles to `global.scss` instead in the docs.

> The docs say to set stacked to true for inline placement. Think you meant false. So stacked is by default true which overlays the headline on top of the hero media.

Fixed this typo to `stacked: false` in the docs.

pkd2512 commented:
I looked at the new code and am happy with the `stacked` and `inline` logic. 

Re: Logic for using photo as background vs inline:
When using an image as a backdrop, it should always be a background image so that it covers the background at all times. So it cannot be used with a `FeaturePhoto`. So, I'd remove the example of using one in `stacked:true` mode as the image resizes and leaves undesirable gaps in the background. If for some reason one needs to use a FeaturePhoto for hero, it will need a lot of media queries to reposition the headline, in which case, it will be prudent to use the inline mode (which is why it was created).

Similarly, when using inline, using a `FeaturePhoto` should be the default implementation as there is no height constraint as such and the developer can using custom logic if needed to resize the image block.

MinamiFunakoshiTR commented:
Thanks for the review, @pkd2512. Here's what changed in the last commit https://github.com/reuters-graphics/graphics-components/pull/246/commits/7b1e56eec6bf94a699f1765127a30809a7cdc854:

- Gets rid of `FeaturePhotoHero` story, the demo that used `FeaturePhoto` as a backdrop image
- Changes the default inline image in `HeroHeadline.svelte` to use `FeaturePhoto` instead of `background-image`.

Would be grateful for a final review. thanks!

hobbes7878 reviewed (approved):
Approved, contingent on @pkd2512's last read through.

pkd2512 commented:
For the inline photo, maybe it's best to not wrap the `FeaturePhoto` in a `GraphicBlock` to avoid repetition of alttext and notes, since the Photo component handles those automatically.

<img width="869" alt="image" src="https://github.com/user-attachments/assets/0dbd7770-4bdb-452d-a299-f8e226a2f1d0" />

Remaining examples look good!
